File: 2932-maximum-strong-pair-xor-i/2932-maximum-strong-pair-xor-i.c
int maximumStrongPairXor(int* nums, int numsSize) {
    int ans = 0;

    for(int i = 0; i < numsSize; i++)
    {
        for(int j = i; j < numsSize; j++)
        {
            if(abs(nums[i] - nums[j]) <= (nums[i] < nums[j] ? nums[i] : nums[j]))
            {
                if((nums[i] ^ nums[j]) > ans)
                    ans = nums[i] ^ nums[j];
            }
        }
    }

    return ans;
}

File: 2932-maximum-strong-pair-xor-i/2932-maximum-strong-pair-xor-i.py
class Solution:
    def maximumStrongPairXor(self, nums: List[int]) -> int:
        nums.sort()
        ans = 0
        n = len(nums)
        
        # Iterate through every possible pair (i, j)
        for i in range(n):
            for j in range(i, n):
                x, y = nums[i], nums[j]
                
                
                if y <= 2 * x:
                    
                    ans = max(ans, x ^ y)
                else:
                    
                    break
                    
        return ans